docs/book/src/reference/creating-events.md

```diff
@@ -0,0 +1,73 @@
+# Creating Events
+
+It is often useful to publish *Event* objects from the controller Reconcile function. 
+
+Events allow users to see what is going on with a particular object, and allow automated processes to see and respond to them.
+
+ Recent Events for an object can be viewed by running `$ kubectl describe <resource kind> <resource name>`. Also, they can be checked by running `$ kubectl get events`. 
+
+<aside class="warning">
+<h1>Events should be raised in certain circumstances only</h1>
+
+Be aware that it is **not** recommended to emit Events for all Operations. If authors raise too many events, it brings bad UX experiences for those consuming the solutions on the cluster, and they will not have attention to what they must be notified of. For more information, please take a look at the [Kubernetes APIs convention][Events]. 
+
+</aside>
+
+## Writing Events
+
+Anatomy of an Event:
+
+```go
+Event(object runtime.Object, eventtype, reason, message string)
+```
+
+- `object` is the object this event is about.
+- `eventtype` is this event type, and is either *Normal* or *Warning*.
+- `reason` is the reason this event is generated. It should be short and unique with `UpperCamelCase` format. The value could appear in *switch* statements by automation.
+- `message` is intended to be consumed by humans.
```

Comment:
Can we add links for each of the type specifications so that its easier. For example:
event type would be https://github.com/kubernetes/api/blob/6c11c9e4685cc62e4ddc8d4aaa824c46150c9148/core/v1/types.go#L6019-L6024
reason would be https://github.com/kubernetes/api/blob/6c11c9e4685cc62e4ddc8d4aaa824c46150c9148/core/v1/types.go#L6048 
message would be https://github.com/kubernetes/api/blob/6c11c9e4685cc62e4ddc8d4aaa824c46150c9148/core/v1/types.go#L6053

Reply:
@Sajiyah-Salat just add the links in the above statements like; 

[(More info)][link-alias] after the explanation I think that would be enough.

Otherwise, all is great for me here and this one has my lgtm

GREAT WORK

Reply:
Can you please squash all commits?
git rebase -i HEAD~<number of commits>
Then, you replace pick with s to squash the commit with the previous one.